util/network-devp2p/src/discovery.rs

```diff
@@ -79,23 +74,21 @@ impl NodeBucket {
 	}
 }
 
-struct Datagramm {
-	payload: Bytes,
-	address: SocketAddr,
+pub struct Datagramm {
+	pub payload: Bytes,
+	pub address: SocketAddr,
 }
 
 pub struct Discovery {
 	id: NodeId,
 	id_hash: H256,
 	secret: Secret,
 	public_endpoint: NodeEndpoint,
-	udp_socket: UdpSocket,
-	token: StreamToken,
 	discovery_round: u16,
 	discovery_id: NodeId,
 	discovery_nodes: HashSet<NodeId>,
 	node_buckets: Vec<NodeBucket>,
-	send_queue: VecDeque<Datagramm>,
+	pub send_queue: VecDeque<Datagramm>,
```

Comment:
I would be in favour of adding methods to `Discovery` to pop from the send queue instead of making this field public, but I don't have a strong opinion on this.